executor/src/witgen/machines/block_machine.rs

```diff
@@ -50,38 +50,35 @@ impl<T: FieldElement> BlockMachine<T> {
         global_range_constraints: &BTreeMap<&PolynomialReference, RangeConstraint<T>>,
     ) -> Option<Box<Self>> {
         for id in connecting_identities {
-            if let Some(sel) = &id.right.selector {
-                // TODO we should check that the other constraints/fixed columns are also periodic.
-                if let Some((selector, period)) = try_to_boolean_periodic_selector(sel, fixed_data)
-                {
-                    let mut machine = BlockMachine {
-                        block_size: period,
-                        selector: selector.clone(),
-                        identities: identities.iter().map(|&i| i.clone()).collect(),
-                        data: witness_cols
-                            .iter()
-                            .map(|n| (n.poly_id() as usize, vec![]))
-                            .collect(),
-                        row: 0,
-                        range_constraints: Default::default(),
-                        global_range_constraints: global_range_constraints
-                            .iter()
-                            .filter_map(|(n, c)| {
-                                n.is_witness().then_some((n.poly_id() as usize, c.clone()))
-                            })
-                            .collect(),
-                        degree: fixed_data.degree,
-                        processing_sequence_cache: ProcessingSequenceCache::new(
-                            period,
-                            identities.len(),
-                        ),
-                    };
-                    // Append a block so that we do not have to deal with wrap-around
-                    // when storing machine witness data.
-                    machine.append_new_block(fixed_data.degree).unwrap();
+            // TODO we should check that the other constraints/fixed columns are also periodic.
+            if let Some(period) = try_to_boolean_periodic_selector(&id.right.selector, fixed_data) {
```

Comment:
But now the function only returns the period and not the selector. Maybe rename?

Reply:
Done!